_posts/en/newsletters/2020-07-29-newsletter.md

```diff
@@ -0,0 +1,236 @@
+    announce transactions they wouldn't accept themselves, so only a
+    disruptive peer that wanted to waste its own upload bandwidth would
+    advertise invalid or unwanted transactions.  However, one type of
+    unwanted transaction today are spends of v1 segwit UTXOs---the types
+    of spends the [BIP341][] specification of [taproot][topic taproot]
+    plans to use.  If taproot activates, this means newer taproot-aware
+    nodes will advertise taproot spends to older taproot-unaware nodes.
+    Each time one those taproot-unaware nodes receives a
+    taproot-spending transaction, it will download it, realize it uses
+    v1 segwit, and throw it away.  This could be very wasteful of
+    network bandwidth, both for older taproot-unaware nodes and newer
+    taproot-aware nodes.  This same problem applies to other proposed
+    changes to network relay policy.
+
+    The solution implemented in this merged PR is to announce
+    transactions by their wtxid---which includes a commitment to the
+    witness data for segwit transactions.  A taproot implementation in
+    Bitcoin Core (see [PR #17977][Bitcoin Core #17977]) could then only
+    relay transactions by their wtxid to prevent newer nodes from
+    accidentally spamming older nodes.
+
+    However, after this PR was merged into Bitcoin Core's master
+    development branch, it was [discussed][meeting xscript] during the
+    weekly Bitcoin Core Development Meeting whether taproot's soft
+    dependency on wtxid relay will make it more complicated to backport
+    taproot to the current 0.20.x branch of Bitcoin Core.  Four options
+    were mentioned during the meeting and in subsequent discussions:
+
+    1. **Backport wtxid:** both wtxid relay and taproot will be
+       backported if there's a 0.20.x taproot release.  John Newbery has
+       already created a quick [wtxid relay backport][].
+
+    2. **Don't backport wtxid:** only backport taproot and just accept that
+       transaction announcements will use more bandwidth than usual
+       until everyone has upgraded to wtxid-using nodes.
+
+    3. **Don't relay taproot:** only backport taproot but don't enable
+       relaying of taproot transactions on backported nodes.  This
+       prevents the immediate bandwidth waste but it may make it harder to
+       get taproot-spending transactions to miners and will reduce the
+       speed and efficiency of [BIP152][] compact blocks.  Worse compact
+       block performance may temporarily increase the number of stale
+       blocks that miners create (especially since the [public FIBRE
+       network][] has recently shut down).
+
+    4. **Don't backport anything:** don't backport wtxid relay or
+       taproot---let taproot wait until after the release of Bitcoin
+       Core 0.21, roughly [expected][Bitcoin Core #18947] in December
+       2020.
+
+    No clear conclusion on which of these options to follow has been
```

Comment:
Maybe: "At the time of this writing, no consensus was reached yet on which option to follow."